content/packages-and-modules/introduction-to-packages-and-modules/about-packages-and-modules.mdx

```diff
@@ -37,7 +37,7 @@ Git URLs used for npm packages can be formatted in the following ways:
 - `git+https://user@hostname/project/blah.git#commit-ish`
 
 The `commit-ish` can be any tag, sha, or branch that can be supplied as
-an argument to `git checkout`. The default `commit-ish` is `master`.
+an argument to `git checkout`. The default `commit-ish` is `main`.
```

Comment:
it's a bit more complex than that, since a) repos from before the change could still have `master`, and b) you can set any default branch you want. what does npm actually do here?

Reply:
There is no default. The default is to not use one and let github.com default it.  When a url requires one (such as with the tarball url) `HEAD` is used.


```javascript
> require('hosted-git-info').fromUrl('github:npm/cli').https()
'git+https://github.com/npm/cli.git'
> require('hosted-git-info').fromUrl('github:npm/cli').tarball()
'https://codeload.github.com/npm/cli/tar.gz/HEAD'
```